### PR TITLE
refacto Grid and Period

### PR DESCRIPTION
**Pull request by flyIchtus**
Gather the Grid and Period objects on Titan and POESY, small sync between APIs, put the objects on base.py

### CODE REVIEW

**Comment by flyIchtus** [comment]
tested on POESY too, via small modification of config file keywords

File: py4cast/datasets/base.py [hunk 1]
```diff
@@ -678,6 +684,131 @@ def summary(self):
...
+    name: str
+
+    def __init__(self, start: int, end: int, step: int, name: str):
+        self.start = dt.datetime.strptime(str(start), "%Y%m%d%H")
+        self.end = dt.datetime.strptime(str(end), "%Y%m%d%H")
+        self.step = step
+        self.name = name
+
+    @property
+    def date_list(self):
```
Comment (cbovalo) [inline py4cast/datasets/base.py:701]: Does this list require to be an array of datetime.datetime? I mean do you have to use pandas just to do this?
If no, maybe you could use `numpy.timedelta64`.
Something like this is possible :
`return np.arange(self.start, self.end, np.timedelta64(self.step, "h"), dtype='datetime64[s]').tolist()`

File: py4cast/datasets/poesy.py [hunk 2]
```diff
@@ -58,118 +59,17 @@ def get_weight(level: float, level_type: str) -> float:
...
-            self.lat[0, 0],  # min y
-            self.lat[-1, -1],  # max y
-        ]
-
-    @cached_property
-    def projection(self):
-        # Create projection
-        return cartopy.crs.LambertConformal(central_longitude=2, central_latitude=46.7)
+def load_poesy_grid_info(grid: Grid) -> GridConfig:
+    geopotential = np.load(SCRATCH_PATH / OROGRAPHY_FNAME)
```
Comment (cbovalo) [inline py4cast/datasets/poesy.py:66]: Titan has its own var env `PY4CAST_TITAN_PATH`. Could Poesy have its own? And pass `OROGRAPHY_FNAME` and `LATLON_FNAME` as var env also ?

**Review by CorentinSeznec — APPROVED** [review]

File: py4cast/datasets/poesy.py (hunk 2 above)
Comment (flyIchtus) [reply py4cast/datasets/poesy.py:66]: ok, for now if you don't mind I just want to focus on the changes, then we will optimize POESY

**Pull request merged**